implementation/routing/src/routing_manager_base.cpp

```diff
@@ -1202,9 +1202,9 @@ void routing_manager_base::remove_local(client_t _client,
 
         // remove disconnected client from offer service history
         std::set<std::tuple<service_t, instance_t, client_t>> its_clients;
-        for (auto& s : local_services_history_) {
-            for (auto& i : s.second) {
-                for (auto& c : i.second) {
+        for (auto const& s : local_services_history_) {
+            for (auto const& i : s.second) {
+                for (auto const& c : i.second) {
```

Comment:
Prefer `const auto` over `auto const`.

```suggestion
        for (const auto& s : local_services_history_) {
            for (const auto& i : s.second) {
                for (const auto& c : i.second) {
```